src/Makefile.am

```diff
@@ -196,6 +198,11 @@ gen_internal_compose_table_SOURCES = \
     ibuskeyuni.c        \
     $(NULL)
 gen_internal_compose_table_CFLAGS = $(AM_CFLAGS)
+$(GEN_INTERNAL_COMPOSE_TABLE): CC=$(CC_FOR_BUILD)
+$(GEN_INTERNAL_COMPOSE_TABLE): CCLD=$(CC_FOR_BUILD)
+$(GEN_INTERNAL_COMPOSE_TABLE): CFLAGS=$(CFLAGS_FOR_BUILD)
+$(GEN_INTERNAL_COMPOSE_TABLE): CPPFLAGS=$(CPPFLAGS_FOR_BUILD)
+$(GEN_INTERNAL_COMPOSE_TABLE): LDFLAGS=$(LDFLAGS_FOR_BUILD)
 gen_internal_compose_table_LDADD = \
```

Comment:
Do you have your problem for gen-install-compose-table only?
I wonder why you don't think the same issue in emoji-parser and unicode-parser.

Reply:
Oh, you're right. We disabled emoji-parser and unicode-parser when cross-compiling.

Reply:
Probably I think ibus cannot support the multi architectures. libibus is used in ibus/binding/vala, ibus/ui/gtk3, ibus/tools besides ibus/src but I don't know if it makes sense to support the multi architectures for gen-install-compose-table only.
I guess you could put `CC=${CC_FOR_BUILD}` in configure.ac or disable `configure --host` option?

Reply:
Maybe I can look into implementing it similar to how Mesa used to do it (in a few days):
<https://github.com/Mesa3D/mesa/blob/mesa-9.1.6/src/glsl/builtin_compiler/Makefile.am>

Reply:
Should be fixed now.